### DISCUSSION TITLE
FPrime-CLI with GDS on non-standard port

### DISCUSSION THREAD

**Discussion by elankford** [nasa/fprime#2809] [Q&A] [answered]
I am trying to execute an fprime-cli command-send to a GDS running on a non-standard port. The CLI documentation page (https://nasa.github.io/fprime/UsersGuide/gds/gds-cli.html#command-send) says that you can pass a "-p" argument, but I am getting an unrecognized arguments error when I do that. 

How can I pass the "ip-port" of the GDS to a fprime-cli command? For reference, I have the most recent version of Fprime in my project.

**Comment by thomas-bc** [comment]
The arguments should be the same as the one you used to spin up `fprime-gds`, so likely `--tts-port`.

You can see all the options with `fprime-cli command-send --help`

**Reply by elankford** [reply]
I'm running two instances of my gds with the following arguments (different ports for each):  ```fprime-gds -n --dictionary [NAME] --packet-spec [NAME] --zmq --zmq-server --ip-port [PORT] -g none --gui-port [PORT]--tts-port [PORT]```.

When I run ```fprime-cli command-send [CMD] --arguments [ARGS] --dictionary [NAME] --packet-spec [NAME] --zmq --zmq-server --tts-port [PORT]``` the command is sent to the newest GDS, regardless of the tts-port specified.

**Reply by elankford** [reply, marked as answer]
Update: Running without using ZMQ solves the issue.

**Reply by LeStarch** [reply]
The TTS port needs to be the same across all clients of the GDS network.  `--zmq` bypasses this using ZMQ as a distribution layer.  It is the preferred solution and will soon become the default.
